de_web_plugin.cpp

```diff
@@ -1865,6 +1872,23 @@ void DeRestPluginPrivate::gpDataIndication(const deCONZ::GpDataIndication &ind)
                 sensorNode.setManufacturer("PhilipsFoH");
                 sensorNode.setSwVersion("PTM216Z");
             }
+            else if (gpdDeviceId == GpDeviceIdOnOffSwitch && options.byte == 0xC5 && extOptions.byte == 0xF2 && ind.payload().size() == 31)
+            {
+                //For the 1 button switch (0 677 23L or ZLGP17):
+                //Button 1: Device commissioning : options.byte 0xC5 , extOptions.byte 0xF2 , gpdsrcid 5345230 , command 0x02 , Payload 02c5f20164366c366d12f5587972fd1dae3dee65ff1de0f001000004022220
+
+                //For the 2 buttons switch (0 677 24L or ZLGP18):
+                //Button 1: Device commissioning : options.byte 0xC5 , extOptions.byte 0xF2 , gpdsrcid 5375258 , command 0x02 , Payload 02c5f29ddf38cc0baaa63c6d8c31176a7d50892a7f84e35f00000004022220
+                //Button 2: Device commissioning : options.byte 0xC5 , extOptions.byte 0xF2 , gpdsrcid 5375259 , command 0x02 , Payload 02c5f2361cf5a1f812cebb4744edfec5d591f07fe3e2935800000004022220
+                
+                // It use as command
+                // 0x20 GpCommandIdOff
+                // 0x22 GpCommandIdToggle
+
+                sensorNode.setModelId("ZGPSWITCH");
```

Comment:
Is it possible to rename the model id to something unique like `legrand.zgp.switch` to avoid confusion with the Philips Hue Tap switch (`ZGPSWITCH`) and allow for specific button mapping in [Home Assistant integration](https://github.com/home-assistant/core/blob/55b0d562cebc8e86898895e746b1a4bf78617b96/homeassistant/components/deconz/device_trigger.py#L109-L115)?

<img width="467" alt="2021-07-13-16-09-26 Google Chrome coredevice_trigger py at 55b0d562cebc8e86898895e746b1a4bf78617b96 · home-assistantcore" src="https://user-images.githubusercontent.com/954168/125466805-b7f909d5-c3fc-4987-9891-2f118548a415.png">

Reply:
Using now "LEGRANDZGPSWITCH"